[SPARK-40404][DOCS] Add precondition description for `spark.shuffle.service.db.backend` in `running-on-yarn.md` (#37853)
apache/spark: docs/running-on-yarn.md

  <td><code>spark.shuffle.service.db.backend</code></td>
  <td>LEVELDB</td>
  <td>

>     When Yarn NodeManager recovery is enabled, this use to specify the kind of disk-base store used in shuffle 
>     service state store, supports `LEVELDB` and `ROCKSDB` now and `LEVELDB` as default value. 

Reviewer: ```suggestion
    When work-preserving restart is enabled in YARN, this is used to specify the disk-base store used in shuffle 
    service state store, supports `LEVELDB` and `ROCKSDB` with `LEVELDB` as default value. 
```

Author: [d8b39ef](https://github.com/apache/spark/pull/37853/commits/d8b39eff34d8e3cf2382caa377355b6e41ec1eb4) fix this

Author: Thanks for your correction

>     The original data store in `LevelDB/RocksDB` will not be automatically convert to another kind of storage now.

Reviewer: `convert` -> `converted` ?

Reviewer 2: +1 for @mridulm comment. And, could you add additional description about what happens at the runtime when the the store types are mismatched. It's deleted and recreated, right?

Author: The old one will not be deleted, but the new one will be created. When the store type is switched, the directory name will change, for example, from `registeredExecutors.ldb` to `registeredExecutors.rdb`, `YarnShuffleService` will create `registeredExecutors.rdb` if it not exists, but `YarnShuffleService` did not know that `registeredExecutors.ldb` existed, so it will not be deleted

Author: Add `The original data store will be retained and the new type data store will be created when switching storage types.` Is that ok ?

Author: @dongjoon-hyun @mridulm Automatic data format conversion may be a useful feature. I think it is more friendly for migrating stock users to use new features. I have filed a Jira SPARK-40464 and will promote its completion if necessary.

Author: > @dongjoon-hyun @mridulm Automatic data format conversion may be a useful feature. I think it is more friendly for migrating stock users to use new features. I have filed a Jira [SPARK-40464](https://issues.apache.org/jira/browse/SPARK-40464) and will promote its completion if necessary.

also cc @panbingkun , what I discussed with you offline yesterday
